internal/resources/datastore/datastore_storage_config.go

```diff
@@ -133,10 +134,26 @@ func (r *Config) mutate(_ context.Context, tenantControlPlane *kamajiv1alpha1.Te
 		finalizersList.Insert(finalizers.DatastoreSecretFinalizer)
 		r.resource.SetFinalizers(finalizersList.UnsortedList())
 
+		// TODO: remove this after multi-tenancy is implemented for NATS
+		if r.DataStore.Spec.Driver == kamajiv1alpha1.KineNatsDriver {
+			// set username and password to the basicAuth values of the NATS datastore
+			username = r.DataStore.Spec.BasicAuth.Username.Content
+			p, err := r.DataStore.Spec.BasicAuth.Password.GetContent(context.Background(), r.Client)
```

Comment:
You can use the context which is unnamed here since not used previously.

Reply:
ahh yes

*doh*